opennebula/resource_opennebula_service.go

```diff
@@ -379,10 +389,8 @@ func resourceOpennebulaServiceDelete(ctx context.Context, d *schema.ResourceData
 				Summary:  "Failed to recover",
 				Detail:   fmt.Sprintf("service (ID: %s): %s", d.Id(), err),
 			})
+			return diags
```

Comment:
I just want to understand why you're only quitting on recovering error ? (I may miss something)

Reply:
Basically, previously when Delete was called, but the service was in state that Delete couldn't be performed, then Recover(true) was called next, and if that was a success the code exited anyways. So the provider returned error that the service cannot be deleted, but the service was removed anyways and that in turn caused discrepancy between state file and the state in OpenNebula instance.

https://github.com/OpenNebula/one-ee/blob/master/src/oca/go/src/goca/service.go#L107

So now if Delete fails then we try Recover(true) which ensures the removal in a more drastic way. In the end if one of those operations is successful we can continue, if both fail we have no choice to return the error and stop. :thinking: